### PR TITLE
Order independent diff option

### PR DESCRIPTION
**Pull request by jsfuchs**
This provides an order independent diff function. However, the diff output is not particularly smart or useful (sorry, I only needed to know if two objects were different, not how).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -148,53 +148,63 @@ function deepDiff(lhs, rhs, changes, prefilter, path, key, stack) {
...
 
-  var ltype = typeof lhs;
-  var rtype = typeof rhs;
-
-  var ldefined = ltype !== 'undefined' || stack && stack[stack.length - 1].lhs.hasOwnProperty(key);
-  var rdefined = rtype !== 'undefined' || stack && stack[stack.length - 1].rhs.hasOwnProperty(key);
+    var ltype = typeof lhs;
+    var rtype = typeof rhs;
+    var ldefined =ltype !== 'undefined'|| stack && stack[stack.length - 1].lhs.hasOwnProperty(key) ;
+      var rdefined =rtype !== 'undefined'|| stack && stack[stack.length - 1].rhs.hasOwnProperty(key);
```
Comment (Kamilius) [inline index.js]: bad formatting and indentation here + line above

File: index.js [hunk 2]
```diff
@@ -206,53 +216,110 @@ function deepDiff(lhs, rhs, changes, prefilter, path, key, stack) {
...
-      case 'D':
-        arr = arrayRemove(arr, index);
-        break;
-      case 'E':
-      case 'N':
-        arr[index] = change.rhs;
-        break;
+
+    return hash;
+  }function applyArrayChange(arr, index, change) {
```
Comment (Kamilius) [inline index.js]: broken formatting

File: index.js [hunk 3]
```diff
@@ -368,45 +435,55 @@ function applyDiff(target, source, filter) {
...
-        });
-        conflictResolution = null;
-      }
-      return accumulateDiff;
+    orderIndependentDiff: {
+      value: accumulateOrderIndependentDiff,
+      enumerable: true
+    },observableDiff: {
+      value: deepDiff,enumerable: true
     },
```
Comment (Kamilius) [inline index.js]: really bad formatting

**Comment by Kamilius** [comment]
Please correct code formatting everywhere in your changes, as it totally disregards code conventions already used in a code

**Comment by jsfuchs** [comment]
@Kamilius - Looks like my IDE went rogue on me when I was merging in conflicts from before. I think I've gotten the styling back in shape - let me know if there's anything else you need.

**Comment by Kamilius** [comment]
@jsfuchs build is failing here. Can you repair it?

**Pull request merged**